stl/inc/strstream

```diff
@@ -34,7 +34,12 @@ public:
     }; // set if character array ownership given away
     using _Strstate = int;
 
-    explicit __CLR_OR_THIS_CALL strstreambuf(streamsize _Count = 0) {
+    // construct with empty character array
+    __CLR_OR_THIS_CALL strstreambuf() {

```

Comment:
This looks like it might be an ABI break (otherwise why would it be annotated `__CLR_OR_THIS_CALL` ?)

Alternately the `__CLR_OR_THIS_CALL` annotation might be incorrect.

Reply:
It is definitely concerning - the presence of explicit calling conventions strongly indicates separately compiled machinery that we can't mess with. However, in the specific case of `strstreambuf`, I believe that this may be okay. The base class `basic_streambuf` has explicit calling conventions because it is indeed separately compiled. However, I can't find any mentions of `strstream` (including `strstreambuf`) in either the static or dynamic libraries. Therefore, I believe that splitting this constructor should be ABI-safe. Our test to verify the export surface will serve as a double-check.

Reply:
I agree with Stephan's analysis. Should we file a separate issue to remove `__CLR_OR_THIS_CALL` from all of this class's function declarations?

Reply:
I think that would be unnecessarily risky for no benefit. This is strstream, after all - the entire thing is deprecated and rarely changes. Also, the virtuals need explicitly specified calling conventions to match the base class, even if the non-virtuals don't.

(Also, vNext will eliminate separately compiled classes.)

Reply:
NO chage done